### PR TITLE
Features/erc20 token

### PR DESCRIPTION
**Pull request by Imad-Allal**
@achab I have made the changes you required.
I don't know how to test the function testApprove correctly, like it seems to be working but I don't know how to send tokens from the owner using the spender address in forge

### CODE REVIEW

**Comment by Imad-Allal** [comment]
This might be the one

File: foundry.toml [hunk 1]
```diff
@@ -2,5 +2,7 @@
 src = "src"
 out = "out"
 libs = ["lib"]
-
+remappings = ["@openzeppelin/=lib/openzeppelin-contracts/"]
+[profile.networks]
+  local = "http://localhost:8545"
```
Comment (achab) [inline foundry.toml:7]: @Imad-Allal why do you need to add this?

File: test/PaymasterToken.t.sol [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    PaymasterToken public paymasterToken;
+    address owner;
+    address spender;
+    address receiver;
+
+    function setUp() public {
+        paymasterToken = new PaymasterToken();
+        owner = address(this);
+        spender = 0x70997970C51812dc3A010C7d01b50e0d17dc79C8;
+        receiver = 0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC;
```
Comment (achab) [inline test/PaymasterToken.t.sol:18]: why using hardcoded addresses here?

File: README.md [hunk 3]
```diff
@@ -39,4 +39,18 @@ After installing OpenZeppelin as a dependency try importing something from it in
...
 forge remappings > remappings.txt
-```
\ No newline at end of file
+```
+
+## Working with Foundry
+### Compiling
+To compile the contracts, run the following command
+```bash
+forge compile
```
Comment (achab) [inline README.md:48]: hmm `forge compile` command does not exist. maybe you meant `forge build`?

**Review by achab — APPROVED** [review]
Great job! Feel free to merge

**Pull request merged**